src/economy/ware_instance.cc

```diff
@@ -196,7 +196,7 @@ WareInstance::WareInstance(DescriptionIndex const i, const WareDescr* const ware
 WareInstance::~WareInstance() {
 	if (supply_) {
 		FORMAT_WARNINGS_OFF
-		molog(kNoTimestamp, "Ware %u still has supply %p\n", descr_index_, supply_);
+		molog(Time(kNoTimestamp), "Ware %u still has supply %p", descr_index_, supply_);
```

Comment:
For this to work we'd need a `logic` dependency in the `base_log` library because we'd need to pass a `const Time&` to the log function and call it's `get()` and `is_valid()` functions. Currently `get()` is expanded in the macro definition so we can do without such a dependency.

Reply:
How about moving time and duration to the base library?